wire/error.go

```diff
@@ -153,6 +153,9 @@ const (
 	// ErrTooManyCFilters is returned when the number of committed filters
 	// exceeds the maximum allowed in a batch.
 	ErrTooManyCFilters
+
+	// ErrInternal is returned for internal package errors.
+	ErrInternal = -1
```

Comment:
If we're going to return an error, please make it explicit.  These types of generic errors always result in poor testability down the line.  Plus, putting this here as -1 means it breaks iota for anything that would come after it (meaning anything new would have to be added before it) and that is a nasty foot gun that would likely be very easy to miss since any code would be referencing them by name and thus unlikely to notice the duplicated error code(s).

Reply:
the -1 is an intentional hint to put new errors above it, not below it.

"internal" is jut the error code. the actual description does contain lots of relevant detail.

but if you'd rather we just don't add any new error codes at all i'm even more alright with that...